stl/inc/xstring

```diff
@@ -161,11 +161,17 @@ struct _Char_traits { // properties of a string or stream element
     }
 
     static _CONSTEXPR20 _Elem* assign(
-        _Out_writes_all_(_Count) _Elem* const _First, size_t _Count, const _Elem _Ch) noexcept
-    /* strengthened */ {
+        _Out_writes_all_(_Count) _Elem* const _First, size_t _Count, const _Elem _Ch) noexcept /* strengthened */ {
         // assign _Count * _Ch to [_First, ...)
         for (_Elem* _Next = _First; _Count > 0; --_Count, ++_Next) {
-            *_Next = _Ch;
+#ifdef __cpp_lib_is_constant_evaluated
+            if (_STD is_constant_evaluated()) {
+                _STD construct_at(_Next, _Ch);

```

Comment:
Urgh, so this breaks the `misc_constexpr` tests, because they assume `_CONSTEXPR20`

However, this is only valid if `_First` points to initialized memory. But for uninitialized memory we need `construct_at` which we only get with `_CONSTEXPR20_DYNALLOC`.

I would consider this a but in the current implementation and move it to `_CONSTEXPR20_DYNALLOC` and guard the current tests appropriately.

Reply:
We should have enabled `_CONSTEXPR20_DYNALLOC` as `constexpr` for both Clang and EDG - my attempt to enable `constexpr` dynamic allocation for everything except MSVC in `<yvals_core.h>` was incomplete 😬 
I've fixed it up and pushed changes 😇